cloudfront/iiif/main.tf

```diff
@@ -0,0 +1,439 @@
+locals {
+  subdomain_modifier = var.environment == "prod" ? "" : "-${var.environment}"
+
+  distro_alias = "iiif${local.subdomain_modifier}.wellcomecollection.org"
+
+  dds_domain   = "dds${local.subdomain_modifier}.dlcs.io"
+  iiif_domain  = "iiif${local.subdomain_modifier}.dlcs.io"
+  loris_domain = "iiif-origin.wellcomecollection.org"
+  dlcs_domain  = "dlcs.io"
+}
+
+resource "aws_cloudfront_distribution" "iiif" {
+  aliases = [
+    local.distro_alias
+  ]
+
+  origin {
+    domain_name = local.dds_domain
+    origin_id   = "dds"
+
+    custom_origin_config {
+      origin_protocol_policy = "match-viewer"
+      origin_ssl_protocols = ["TLSv1"]
+
+      http_port = 80
+      https_port = 443
+    }
+  }
+
+  origin {
+    domain_name = local.dlcs_domain
+    origin_id   = "dlcs"
+
+    custom_origin_config {
+      origin_protocol_policy = "match-viewer"
+      origin_ssl_protocols = ["TLSv1"]
+
+      http_port = 80
+      https_port = 443
+    }
+  }
+
+  origin {
+    domain_name = local.loris_domain
+    origin_id   = "loris"
+
+    custom_origin_config {
+      origin_protocol_policy = "match-viewer"
+      origin_ssl_protocols = ["TLSv1"]
+
+      http_port = 80
+      https_port = 443
+    }
+  }
+
+  origin {
+    domain_name = local.iiif_domain
+    origin_id   = "iiif"
+
+    custom_origin_config {
+      origin_protocol_policy = "match-viewer"
+      origin_ssl_protocols = ["TLSv1"]
+
+      http_port = 80
+      https_port = 443
+    }
+  }
```

Comment:
It might be overkill here, but it occurs to me that this would be a good use for dynamic blocks in Terraform.